### DISCUSSION TITLE
v3.4.3

### DISCUSSION THREAD

**Discussion by danielroe** [nuxt/nuxt#20568] [Announcements]
> 3.4.3 is a patch release with the latest bug fixes. 🐞 It is expected that the next release will be v3.5, in approximately two weeks' time.

## ✅ Upgrading

As usual, our recommendation for upgrading is to run:

```sh
nuxi upgrade --force
```

This will refresh your lockfile as well, and ensures that you pull in updates from other dependencies that Nuxt relies on, particularly in the unjs ecosystem.

## 👉 Changelog

[compare changes](https://github.com/nuxt/nuxt/compare/v3.4.2...v3.4.3)

### 🩹 Fixes

  - **nuxt:** Don't call `timeEnd` unless we're debugging ([#20424](https://github.com/nuxt/nuxt/pull/20424))
  - **nuxt:** Use key to force server component re-rendering ([#19911](https://github.com/nuxt/nuxt/pull/19911))
  - **nuxt:** Add basic typings for `<ClientOnly>` ([f1ded44e8](https://github.com/nuxt/nuxt/commit/f1ded44e8))
  - **nuxt:** Use `event.node.req` in cookie utility ([#20474](https://github.com/nuxt/nuxt/pull/20474))
  - **deps:** Relax upper node version constraint ([#20472](https://github.com/nuxt/nuxt/pull/20472))
  - **nuxi,schema:** Support `devServer.https: true` ([#20498](https://github.com/nuxt/nuxt/pull/20498))
  - **nuxt:** Throw 404 when accessing `/__nuxt_error` directly ([#20497](https://github.com/nuxt/nuxt/pull/20497))
  - **nuxt:** Use `callAsync` for executing hooks with context ([#20510](https://github.com/nuxt/nuxt/pull/20510))
  - **nuxt:** Improved typing support for app config ([#20526](https://github.com/nuxt/nuxt/pull/20526))
  - **nuxt:** Call `app:error` in SSR before rendering error page ([#20511](https://github.com/nuxt/nuxt/pull/20511))
  - **nuxt:** Restrict access to single renderer outside of test/rootDir ([#20543](https://github.com/nuxt/nuxt/pull/20543))
  - **nuxt:** Throw errors when running legacy `asyncData` ([#20535](https://github.com/nuxt/nuxt/pull/20535))
  - **nuxt:** Transform `#components` imports into direct component imports ([#20547](https://github.com/nuxt/nuxt/pull/20547))
  - **nuxt:** Return `RenderResponse` for redirects ([#20496](https://github.com/nuxt/nuxt/pull/20496))

### 📖 Documentation

  - Fix typos on docs homepage ([#20456](https://github.com/nuxt/nuxt/pull/20456))
  - Update links to `vue-router` docs ([#20454](https://github.com/nuxt/nuxt/pull/20454))
  - Remove RC reference and add link to vue migration build ([#20485](https://github.com/nuxt/nuxt/pull/20485))
  - Add cdn proxy section ([#20408](https://github.com/nuxt/nuxt/pull/20408))
  - Add a next steps link to first page of migration docs ([#20512](https://github.com/nuxt/nuxt/pull/20512))
  - Add custom fetch composable example ([#20115](https://github.com/nuxt/nuxt/pull/20115))
  - Adjust wrong link after repo migration ([#20536](https://github.com/nuxt/nuxt/pull/20536))

### ✅ Tests

  - Exclude default error pages from bundle test ([ba320a558](https://github.com/nuxt/nuxt/commit/ba320a558))
  - Update snapshot ([ac598e4f4](https://github.com/nuxt/nuxt/commit/ac598e4f4))

### 🤖 CI

  - Publish edge release with provenance ([ec1503731](https://github.com/nuxt/nuxt/commit/ec1503731))
  - Run release script with node 18 ([0d10e9734](https://github.com/nuxt/nuxt/commit/0d10e9734))
  - Try releasing `nuxt-edge` with provenance ([753c4c2a3](https://github.com/nuxt/nuxt/commit/753c4c2a3))
  - Run nuxt2 nightly release on node 14 again ([48c034cf0](https://github.com/nuxt/nuxt/commit/48c034cf0))

<hr /><em>This discussion was created from the release <a href='https://github.com/nuxt/nuxt/releases/tag/v3.4.3'>v3.4.3</a>.</em>

**Comment by jd1378** [comment]
@danielroe [This new code](https://github.com/nuxt/nuxt/blob/1464294d6abe13b54fdb8d204732345551c8e647/packages/nuxt/src/components/transform.ts#L32) has broken the functionality over at [nuxt-svgo](https://github.com/cpsoinos/nuxt-svgo)
the reason is that nuxt-svgo has a custom vite plugin that uses the import url query to determine what to do with the svg file
this new code now appends `?component=async` at the end of a url that already has `?componentext` (there's other possible queries as well) , which breaks it by combining into (`?componentext?component=async`)

the fault is mine I guess ? because I relied on some implementation detail rather than API, but I couldn't find an API for adding query to import urls when using `addComponentsDir`

**Reply by danielroe** [reply]
Thanks for letting me know. Let's fix this, either in Nuxt or at nuxt-svgo.

cc: @antfu

**Reply by jd1378** [reply]
I don't know how this can affect other packages, but it looks like it would be nice to fix this at Nuxt as well

I'm working on a fix for `nuxt-svgo`, I simply need to remove anything I'm doing to the `filePath` in `extendComponent` of `addComponentsDir`. just checking if there's some way to make it a non-breaking change on my end

**Reply by jd1378** [reply]
I have released the fix at nuxt-svgo

**Reply by CastroM081297** [reply]
> @danielroe [This new code](https://github.com/nuxt/nuxt/blob/1464294d6abe13b54fdb8d204732345551c8e647/packages/nuxt/src/components/transform.ts#L32) has broken the functionality over at [nuxt-svgo](https://github.com/cpsoinos/nuxt-svgo)
> the reason is that nuxt-svgo has a custom vite plugin that uses the import url query to determine what to do with the svg file
> this new code now appends `?component=async` at the end of a url that already has `?componentext` (there's other possible queries as well) , which breaks it by combining into (`?componentext?component=async`)
> 
> the fault is mine I guess ? because I relied on some implementation detail rather than API, but I couldn't find an API for adding query to import urls when using `addComponentsDir`

Ok

**Reply by antfu** [reply]
Fix is here: https://github.com/nuxt/nuxt/pull/20591

**Reply by oleghalin** [reply]
Even with @jd1378 fix i still cannot use ?component query in import

**Reply by jd1378** [reply]
@oleghalin create an issue on nuxt-svgo repo and provide some details, we'll track it there

maybe it's something with `?component` query, because I can definitely use `?skipsvgo` in import without any problem
overall we probably have to wait for nuxt's fix release to see how it goes
